test/jest/pre-init-utils.spec.ts

```diff
@@ -0,0 +1,56 @@
+import { processReadyQueue } from '../../src/pre-init-utils';
+
+describe('pre-init-utils', () => {
+    describe('#processReadyQueue', () => {
+        it('should return an empty array if readyQueue is empty', () => {
+            const result = processReadyQueue([]);
+            expect(result).toEqual([]);
+        });
+
+        it('should process functions passed as arguments', () => {
+            const functionSpy = jest.fn();
+            const readyQueue: Function[] = [functionSpy, functionSpy, functionSpy];
+            const result = processReadyQueue(readyQueue);
+            expect(functionSpy).toHaveBeenCalledTimes(3);
+            expect(result).toEqual([]);
+        });
+
+        it('should process functions passed as arrays', () => {
+            const functionSpy = jest.fn();
+            (window.mParticle as any) = {
+                fakeFunction: functionSpy,
+            };
+            const readyQueue = [['fakeFunction']];
+            processReadyQueue(readyQueue);
+            expect(functionSpy).toHaveBeenCalled();
+        });
+
+        it('should process functions passed as arrays with arguments', () => {
+            const functionSpy = jest.fn();
+            (window.mParticle as any) = {
+                fakeFunction: functionSpy,
+                args: () => {},
```

Comment:
Is this `args` function necessary? I haven't pulled this code down, but as I read through the `processReadyQueue` function, the only times I see `args` is where it references the item in the array, as opposed to it being a key on `window.mParticle`.

Reply:
I'm open to discussing this further. Adding `args` here was the only way I could get it to hit that specific codepath in my tests.

Reply:
let's hop on a zoom about this to walk through

Reply:
hm. when i comment out line 32, it passes for me.  see screenshot below:
![Uploading image.png…]()

Reply:
It's weird but commenting out `arg` goes to a different code path that has the same result. It's likely a redundant conditional and could use some clean up. We should definitely sync up on the use cases to make sure nothing is missed.